pkg/cloudprovider/types.go

```diff
@@ -44,6 +43,11 @@ type CloudProvider interface {
 	Validate(context.Context, *v1alpha5.Constraints) *apis.FieldError
 	// Name returns the CloudProvider implementation name.
 	Name() string
+	// ResourcePriorities provides information to bin-packing regarding the sort priority of resources. If CPU is before memory
+	// in this list, the bin-packer will prefer to schedule on nodes with less CPU than memory.  This should always include at
+	// least v1.ResourceMemory and v1.ResourceCPU, however it also is suggested to include any specific resource types
+	// (e.g GPUs) ordered before CPU and memory.
+	ResourcePriorities() []v1.ResourceName
```

Comment:
This is used to provide an orderings for:

-  {cpu: 2, mem: 8GB, nvidia: 1} <  {cpu: 2, mem: 8GB, amd: 1} # so you can define a consistent ordering for disparate resource types
- {cpu: 8, mem: 32Gb} < { cpu:1 mem: 1GB, nvidia: 1} # prefer instance types with more cpu/memory over those with special resources

Reply:
Could we expose this as a `Compare` func?